sockeye/attention.py

```diff
@@ -310,6 +310,9 @@ def attend(att_input: AttentionInput, att_state: AttentionState) -> AttentionSta
                                               num_hidden=self.num_hidden,
                                               no_bias=True, name="%squery_hidden_fc" % self.prefix)
 
+            # scale down dot product by sqrt(num_hidden) [Vaswani et al, 17]
+            query *= self.num_hidden ** -0.5
```

Comment:
ah yes, you are right I haven't thought about backwards compatibility here. I'll expose another type dot_scaled that maps to a parameter passed to DotAttention.